node/service/src/relay_chain_selection.rs

```diff
@@ -338,16 +338,27 @@ impl<B> SelectChain<PolkadotBlock> for SelectRelayChain<B>
 				.map_err(|e| ConsensusError::Other(Box::new(e)))?
 			{
 				// No approved ancestors means target hash is maximal vote.
-				None => (target_hash, target_number),
-				Some((s_h, s_n)) => (s_h, s_n),
+				None => (target_hash, target_number, Vec::new()),
+				Some((s_h, s_n, s_bd)) => (s_h, s_n, s_bd),
 			}
 		};
 
+		// 3. Constrain according to disputes:
+		let (tx, rx) = oneshot::channel();
+		overseer.send_msg(DisputeCoordinatorMessage::DetermineUndisputedChain{
+				base_number: subchain_number,
```

Comment:
I believe `base` here should be `target_number`. That is, the number of `target_hash` passed into `fn finality_target`. `subchain_number` represents the head of the finalizable chain whereas `target_number` represents the block we _must_ finalize.

`DetermineUndisputedChain` also requires that the `block_descriptions` passed to it represent a complete chain from `base_number + 1 ..= approved_subchain_head`.

Reply:
So we might also want a sanity check that `subchain_block_descriptions.len() == subchain_number - base_number` before passing anything into `DetermineUndisputedChain`. If that check fails, it indicates a logic error, so we should vote on the minimal block.